### PR TITLE
Docker setup

### PR DESCRIPTION
**Pull request by chriseth**
Extracted from https://github.com/chriseth/pathfinder2/pull/16

### CODE REVIEW

File: .github/workflows/dev.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: Build and push the dev image
+
+on:
+  push:
+    branches: [ feature/dockerfile ]
```
Comment (chriseth) [inline .github/workflows/dev.yml]: @jaensen is this still correct as is?

File: .github/workflows/build-and-push.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        driver-opts: |
+          image=moby/buildkit:v0.10.6
+
+    -
+      name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v4
+      with:
+        images: |
+          ${{ vars.GC_REGISTRY }}/${{ vars.GC_PROJECT_ID }}/${{ inputs.image }}
```
Comment (chriseth) [inline .github/workflows/build-and-push.yml]: Are these variables I need to set somewhere?

File: .github/workflows/dev.yml (hunk 1 above)
Comment (jaensen) [reply .github/workflows/dev.yml]: Hi! These are the workflows for our docker image build. They'll most likely won't work outside of our GitHub organization :/
You can either ignore/remove it or we can work together with @almereyda to setup the stuff in a way that will also work from your repository.

File: .github/workflows/build-and-push.yml (hunk 2 above)
Comment (jaensen) [reply .github/workflows/build-and-push.yml]: same as above

**Comment by jaensen** [comment]
I think for now it's no problem to remove the docker-build related stuff (just keep the dockerfile around).
If you're fine with it, I think it would make most sense to transfer the ownership of the repo to the CirclesUBI-Orga so that we can publish the pathfinder along with all other components.

**Comment by almereyda** [comment]
I agree! From a maintenance perspective, as there are build workflows, image registries and third party dependencies on these images in various places (land-loca, cloud deployments, circles.garden, group currency app, etc.) it makes sense to consolidate on a single source of truth for the pathfinder, and distribute its artefacts from there.

@chriseth Would you be open to transferring ownership of this repository here, chriseth/pathfinder2, into the CirclesUBI organisation, in so we can continue to maintain it from there, with all the goodies introduced by our custom feature brances, and as such, custom fork?

**Comment by chriseth** [comment]
Sure, no worries, let's transfer it! It doesn't have a frontend like the old pathfinder does.

This might also be a good point to publish a cargo package, but let's do the transfer first.

**Comment by chriseth** [comment]
I tried to transfer it,but got the following error:
```
CirclesUBI/pathfinder2 already exists and You don’t have the permission to create public repositories on CirclesUBI
```

**Comment by almereyda** [comment]
This also includes the changes from #17, as the default branch has switched from `main` to `dev` ¹, and that change was not included in rolling `dev` here, yet. 72cfb6f

The three merges were needed after rebasing the `docker` branch on the current `dev`, only after locally updating the new `dev` with the state from the old `dev` (now in the `jaensen` namespace). 8e9e27c

Why it became necessary to pull the original `docker` branch on top of the newly reshaped local `docker` branch. 53d93a0

¹ https://github.com/CirclesUBI/pathfinder2/settings/branches

**Comment by almereyda** [comment]
OP-89

**Pull request merged**